VS_Project/Pokerapp/Pokerapp/AdminPage.cshtml

```diff
@@ -19,32 +19,53 @@
 
     int number = 1;         //later toevoegen table number
     int rebuy = 0;          //aantal rebuys standaard op 0
-
+    string error = "";
     Validation.Add("name", Validator.StringLength(40, 1, "Name has to between 1 - 40"));
-    Validation.RequireField("name", "A name is Required");
+
+
+
 
     if (IsPost)
     {
+        dynamic room = db.QuerySingle("SELECT * FROM Room WHERE Room_Code = @0", roomCode);
+        double maxplayerstable = room.Tbl_Max_Size;
+        double minplayerstable = room.Tbl_Min_Size;
+        double maxtables = room.Tbl_Amount;
+        double maxplayerstotal = maxplayerstable * maxtables;
+        int playercount = db.QueryValue("SELECT Count(Name) FROM Players WHERE Room_Code = @0", roomCode);
 
-        if (Validation.IsValid())
+
+        string name = Request["name"];
+        if (Validation.IsValid() && Request["AddName"] == "Add" && db.QueryValue("SELECT Name FROM Players WHERE Name = @0 AND Room_Code = @1", name, roomCode) is null)
         {
-            string name = Request["name"];
-            if (Request["AddName"] == "Add" && db.QueryValue("SELECT Name FROM Players WHERE Name = @0 AND Room_Code = @1", name, roomCode) is null)
+            if (playercount < maxplayerstotal)
             {
                 db.Execute("INSERT INTO Players (Name, Room_Code, Tbl_nr, Rebuy) VALUES (@0, @1, @2, @3)", name, roomCode, number, rebuy);
                 name = null;
             }
-            else if (Request["delete"] == "delete")
+            else
             {
-                db.Execute("DELETE FROM Players WHERE Name = @0 AND Room_Code = @1", Request.Form["rownameform"], roomCode);
+                error = "You cant add more then " + maxplayerstotal + " players";
             }
         }
+
+        if (Request["delete"] == "delete")
+        {
+            db.Execute("DELETE FROM Players WHERE Name = @0 AND Room_Code = @1", Request.Form["rownameform"], roomCode);
+        }
+
         if (Request["Proceed"] == "Proceed")
         {
-            Shuffle shuffle = new Shuffle();
-            shuffle.ShuffleTable(db, roomCode);
-            db.Execute("UPDATE Room SET Started_Time = @0 WHERE Room_Code = @1", DateTime.Now, roomCode);
-            Response.Redirect("~/PlayingAdmin/" + roomCode);
+            if(playercount > 2 && playercount <= maxplayerstotal)
+            {
+                Shuffle shuffle = new Shuffle();
+                shuffle.ShuffleTable(db, roomCode);
+                Response.Redirect("~/PlayingAdmin/" + roomCode);
+            }
+            else
+            {
+                error = "Add atleast 3 players!";
```

Comment:
Misschien kun je hier een @(min_size +1) van maken bij de 3, voor het geval dat de min_size op 4 of zo wordt gezet